File: freshmart/src/types/order.ts
export type PaymentMethod = 'Cash' | 'VietQR' | 'PosCard' | 'cash' | 'vietqr' | 'pos_card';
export type OrderStatus = 'Pending' | 'Completed' | 'Cancelled' | 'pending' | 'completed' | 'cancelled';

export interface OrderItem {
  id: string;
  orderId: string;
  productId: string;
  productName: string;
  sku: string;
  quantity: number;
  unitPrice: number;
  costPrice: number;
  discountPercent: number;
  lineTotal: number;
  createdAt: string;
}

export interface Order {
  id: string;
  code: string;
  shiftId: string | null;
  shiftName: string | null;
  customerId: string | null;
  customerName: string;
  customerPhone: string | null;
  cashierId: string | null;
  cashierName: string;
  subtotal: number;
  discountPercent: number;
  discountAmount: number;
  vatAmount: number;
  total: number;
  paymentMethod: PaymentMethod;
  amountReceived: number;
  changeAmount: number;
  status: OrderStatus;
  note: string | null;
  createdAt: string;
  updatedAt: string;
  items: OrderItem[];
}

export interface OrderItemRequest {
  productId: string;
  productName: string;
  sku: string;
  quantity: number;
  unitPrice: number;
  discountPercent?: number;
  lineTotal: number;
}

export interface CheckoutRequest {
  shiftId?: string;
  customerId?: string;
  customerName?: string;
  customerPhone?: string;
  cashierId?: string;
  cashierName?: string;
  items: OrderItemRequest[];
  subtotal: number;
  discountPercent?: number;
  discountAmount?: number;
  vatRate?: number;
  vatAmount?: number;
  total: number;
  paymentMethod: PaymentMethod;
  amountReceived?: number;
  changeAmount?: number;
  note?: string;
}

export interface OrderFilterParams {
  search?: string;
  shiftId?: string;
  customerId?: string;
  cashierId?: string;
  paymentMethod?: PaymentMethod;
  status?: OrderStatus;
  fromDate?: string;
  toDate?: string;
  page?: number;
  limit?: number;
}

export interface VietQrResponse {
  qrUrl: string;
  bankName: string;
  bankAccount: string;
  accountName: string;
  amount: number;
  transferContent: string;
}

File: Models/Order.cs
using dotnet_backend_freshmart.Models.Enums;

namespace dotnet_backend_freshmart.Models
{
    public class Order
    {
        public Guid Id { get; set; }

        public string Code { get; set; } = string.Empty;

        // Khóa ngoại liên kết tới Ca làm việc thu ngân (Shift)
        public Guid? ShiftId { get; set; }
        public virtual Shift? Shift { get; set; }

        // Khách hàng mua hàng (nullable nếu khách lẻ)
        public Guid? CustomerId { get; set; }
        public string CustomerName { get; set; } = "Khách lẻ vãng lai";
        public string? CustomerPhone { get; set; }

        // Thu ngân thực hiện giao dịch (Cashier)
        public Guid? CashierId { get; set; }
        public virtual Employee? Cashier { get; set; }
        public string CashierName { get; set; } = string.Empty;

        // Tiền hàng & Chiết khấu
        public decimal Subtotal { get; set; } = 0;
        public decimal DiscountPercent { get; set; } = 0;
        public decimal DiscountAmount { get; set; } = 0;
        public decimal VatAmount { get; set; } = 0;
        public decimal Total { get; set; } = 0;

        // Thanh toán & Tiền thối
        public PaymentMethod PaymentMethod { get; set; } = PaymentMethod.Cash;
        public decimal AmountReceived { get; set; } = 0;
        public decimal ChangeAmount { get; set; } = 0;

        public OrderStatus Status { get; set; } = OrderStatus.Completed;

        public string? Note { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Danh sách các mặt hàng trong đơn
        public virtual ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
    }
}
